pages/1.10/deploying-services/marathon-on-marathon/advanced/index.md

```diff
@@ -166,211 +186,156 @@ curl -i -k \
-       ```
-     
-1. Copy the `docker.tar.gz` file to the `/home/core` directory of each of your private agents.
+      ```bash
+      dcos security secrets get ${SERVICE_ACCOUNT_SECRET} --json | jq -r .value | jq
+      ```
 
-   ```bash
-   for i in $PRIVATE_AGENT_IPS; do scp -o StrictHostKeyChecking=no docker.tar.gz core@$i:~/docker.tar.gz ; done
-   ```
-# Step 5 - Create a Service Account Secret
-In this step, a secret is created for the Marathon service account and stored in the Secret Store.
+  *  Delete the private key file from your file system to prevent bad actors from using the private key to authenticate to DC/OS.
 
-Create a secret (`<path-to-secret-name>`) for your service account. The secret will contain the private key (`<private-key>.pem`) and the name of the service account (`<service-account-id>`).
 
-### Disabled
+# Step 5: Assign Permissions (Strict mode only)
+In this step, permissions are assigned to the Marathon-on-Marathon instance. Permissions are required in strict mode and are ignored in other security modes.
 
-```bash
-dcos security secrets create-sa-secret <private-key>.pem <service-account-id> <path-to-secret-name>
-```
+| Security Mode | Permissions |
+|---------------|----------------------|
+| Disabled | Not available |
+| Permissive | Not available |
+| Strict | Required |
 
-### Permissive
-    
-```bash
-dcos security secrets create-sa-secret <private-key>.pem <service-account-id> <path-to-secret-name>
-```
+All CLI commands can also be executed via the [IAM API](/1.10/security/ent/iam-api).
 
-### Strict
-    
-```bash
-dcos security secrets create-sa-secret --strict <private-key>.pem <service-account-id> <path-to-secret-name>
-```  
+Grant service account `${SERVICE_ACCOUNT_ID}` permission to launch Mesos tasks that will execute as Linux user `nobody`.
 
-#### Recommendations
+To allow executing tasks as a different Linux user, replace `nobody` with that user's Linux user ID. For example, to launch tasks as Linux user `bob`, replace `nobody` with `bob` below.
 
--  Review your secret to ensure that it contains the correct service account ID, private key, and `login_endpoint` URL. If you're in `strict` it should be HTTPS, in `disabled` or `permissive` mode it should be HTTP. If the URL is incorrect, try [upgrading the DC/OS Enterprise CLI](/1.10/cli/enterprise-cli/#ent-cli-upgrade), deleting the secret, and recreating it. You can use this commands to view the contents:
-   
-   ```bash
-   dcos security secrets list /
-   ```
-   
-   Alternatively, if you have [jq 1.5 or later](https://stedolan.github.io/jq/download) installed, you can use this command:
+Note that the `nobody` and `root` users exist on all agents by default, but if a custom `bob` user is specified it must have been manually created (using the `adduser` or similar utility) on every agent that tasks can be executed on.
```

Comment:
I propose to add a section "Known pitfalls" here which can go something like this:
```
Known pitfalls:
- when launching docker containers user _nobody_ does not have enough rights to execute the command e.g.
stating _nginx_ as _nobody_ will fail because _nobody_ does not have the right  to write logs to /var/log 
(as nginx needs). 
- If using custom users e.g. _bob_, this user has to exist on the agent and within the container
- user `nobody` has different UIDs on different systems (99 on coreos, 65534 on ubuntu). Depending on the
agent's distribution you might need to modify the container image so that UIDs match! (the same goes if 
you use the user _bob_)
- when using a new user e.g. _bob_ don't forget to give Marathon service account permissions to run tasks 
with this user
There an open JIRA ticket to document these pitfalls: https://jira.mesosphere.com/browse/DCOS-26271
```

Reply:
Wait what? It has to exist on the agent _and_ the container?